[stable/datadog] v2.0.0 major release

clamoriniere opened the pull request:
#### What this PR does / why we need it:

* Remove Datadog agent deployment configuration.
* Cleanup resources labels, to fit with recommended labels.
* Cleanup useless or unused values parameters.

Fixes: #20485
Include:  #20460 #20449

#### Special notes for your reviewer:


#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

clamoriniere committed:
[stable/datadog] v2.0.0 cleaning and refactoring

* Remove Datadog agent deployment configuration.
* Cleanup resources labels, to fit with recommended labels.
* Cleanup useless or unused values parameters.

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
[stable/datadog] cleanup refactor RBAC resources

* each component have its own RBAC configuration (create,configuration).
* container runtime socket update values configuration simplification.
* `nameOverride` `fullnameOverride` is now optional in values.yaml

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

rifelpet commented on stable/datadog/README.md:
```diff
@@ -223,9 +223,10 @@ helm upgrade -f datadog-values.yaml <RELEASE_NAME> stable/datadog --recreate-pod
 ### CRI integration
 
-As of the version 6.6.0, the Datadog Agent supports collecting metrics from any container runtime interface used in your cluster. Configure the location path of the socket with `datadog.criSocketPath` and make sure you allow the socket to be mounted into the pod running the agent by setting `datadog.useCriSocketVolume` to `True`.
+As of the version 6.6.0, the Datadog Agent supports collecting metrics from any container runtime interface used in your cluster. Configure the location path of the socket with `datadog.criSocketPath`; default is the Docker container runtime socket. To desactivate this support, you just need to unset the `datadog.criSocketPath` setting.
```
```suggestion
As of the version 6.6.0, the Datadog Agent supports collecting metrics from any container runtime interface used in your cluster. Configure the location path of the socket with `datadog.criSocketPath`; default is the Docker container runtime socket. To deactivate this support, you just need to unset the `datadog.criSocketPath` setting.
```

clamoriniere replied:
👍

clamoriniere committed:
update stable/datadog/ci values.yaml

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
[datadog] fix linter error

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
[datadog] fix missing merge

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

vboulineau reviewed (commented):
I guess README needs to be updated to state that we do need Kube >= 1.9 as you removed checks around apps/v1 usage

clamoriniere committed:
refactor values.yaml

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
update value.yaml

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

L3n41c committed:
Add a table with the correspondence between the v1 and v2 parameters

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

gbbr committed:
templates: enable APM by default

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

gbbr committed:
Leave apm.enabled setting but change default.

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

gbbr committed:
DD_APM_NON_LOCAL_TRAFFIC=true when apm.enabled is true

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

gbbr committed:
Enable hostPort binding 8126 by default when apm.enabled is true.

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

gbbr committed:
values: make apm.enabled default to false

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

gbbr committed:
Address PR comments.

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
refactor dogstatsd parameters

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
update after comments

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
fix dogstatsd parameters path

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

leeavital commented on stable/datadog/templates/system-probe-configmap.yaml:
```diff
@@ -18,24 +13,19 @@ metadata:
-      enable_conntrack : {{ $.Values.systemProbe.enableConntrack }}
-      bpf_debug: {{ $.Values.systemProbe.bpfDebug }}
-{{- if eq .Values.systemProbe.seccomp "localhost/system-probe" }}
+{{- if eq .Values.datadog.systemProbe.seccomp "localhost/system-probe" }}
```
why did the `if` statement move up two lines?

clamoriniere replied:
I think it is a merge issue. fixing it

leeavital commented on stable/datadog/templates/daemonset.yaml:
```diff
@@ -24,85 +14,75 @@ spec:
+        {{- if .Values.agents.customAgentConfig }}
+        checksum/agent-config: {{ tpl (toYaml .Values.agents.customAgentConfig) . | sha256sum }}
         {{- end }}
+        {{- if and .Values.datadog.systemProbe.enabled }}
```
I think the `and` is unnecessary here

leeavital commented on stable/datadog/templates/daemonset.yaml:
```diff
@@ -24,85 +14,75 @@ spec:
         {{- end }}
+        {{- if and .Values.datadog.systemProbe.enabled }}
+        container.apparmor.security.beta.kubernetes.io/system-probe: {{ .Values.datadog.systemProbe.apparmor }}
+        container.seccomp.security.alpha.kubernetes.io/system-probe: {{ .Values.datadog.systemProbe.seccomp }}
```
We should respect the value of Values.systemProbe.seccomp here

clamoriniere committed:
update system-probe manifest after review

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
update ci configuration and fix cluster-agent config

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
change cluster-agent probes endpoint

change the cluster-agent probes endpoint in order to be resilent if
never the datadog api is not reachable.

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
Fix several typo thanks to contributions

Include: #20460 #20449

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
adding new approvers/reviewers

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
[stable/datadog] add resoures field for init-containers

Fixe: #20035
Include: #20461

Add the possibility to overwrite the resources associated to the
init-containers.

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

johnkost commented on stable/datadog/templates/NOTES.txt:
```diff
@@ -60,32 +60,8 @@ agents use the same shared token.
-  We encourage you to migrate to a single DaemonSet.
-{{- end }}
+{{- if .Values.datadog.apm.enabled }}
+The Datadog Agent is listening on port 8126.
```
```suggestion
The Datadog Agent is listening on port {{ .Values.datadog.apm.port }}.
```

johnkost commented on stable/datadog/templates/cluster-agent-deployment.yaml:
```diff
@@ -78,25 +77,29 @@ spec:
         {{- end }}
         env:
+          - name: DD_HEALTH_PORT
+            value: {{ .Values.clusterAgent.healthPort | default 8080 | quote }}
```
For consistency:
```suggestion
            value: {{ default 8080 .Values.clusterAgent.healthPort | quote }}
```

johnkost commented on stable/datadog/templates/cluster-agent-deployment.yaml:
```diff
@@ -78,25 +77,29 @@ spec:
             valueFrom:
               secretKeyRef:
                 name: {{ template "datadog.apiSecretName" . }}
                 key: api-key
```
Can this be made configurable? If using existing secrets to source these, you're required to look at the charts or create the secret and then look at the secret to see the required key

johnkost commented on stable/datadog/templates/cluster-agent-deployment.yaml:
```diff
@@ -78,25 +77,29 @@ spec:
             valueFrom:
               secretKeyRef:
                 name: {{ template "datadog.appKeySecretName" . }}
                 key: app-key
```
Can this be made configurable? If using existing secrets to source these, you're required to look at the charts or create the secret and then look at the secret to see the required key

clamoriniere replied:
the key "app-key" and "api-key" is hardcoded like this since a wild and we didn't received any issue about it.
So I prefer to not introduce a new parameter if everyone is ok like this.
WDYT?

johnkost replied:
That's fine. I only suggested it since this was a major change to the chart.

clamoriniere replied:
can be added after this it will only introduce new parameters

johnkost commented on stable/datadog/templates/container-agent.yaml:
```diff
@@ -1,14 +1,14 @@
-    hostPort: {{ default 8125 .Values.datadog.dogStatsDPort }}
+  - containerPort: {{ .Values.datadog.dogstatsd.port }}
+    {{- if .Values.datadog.dogstatsd.useHostPort }}
+    hostPort: {{ .Values.datadog.dogstatsd.port }}
```
In some places, defaults are provided in the template itself. Others, it relies on the default in the `values.yaml`. Any reasoning for one or the other?

clamoriniere replied:
The only reason it due to the history of this chart. I will move everything in the values.yaml

johnkost commented on stable/datadog/templates/daemonset.yaml:
```diff
@@ -132,18 +112,18 @@ spec:
         - hostPath:
-            path: {{ .Values.datadog.containerLogsPath | quote }}
-          name: logcontainerpath
+            path: /var/lib/docker/containers
```
Can this be pulled into a value to allow it to be overridden if necessary?

clamoriniere replied:
to my knowledge, this is the standard path and it is used by every manager solution and installer.
I prefer to have someone that opens an issue to add another parameter.

johnkost replied:
That's fine. This can be created in an issue if anyone else in the future has problems

johnkost commented on stable/datadog/templates/daemonset.yaml:
```diff
@@ -132,18 +112,18 @@ spec:
         - hostPath:
             path: /var/log/pods
```
Can this be pulled into a value to allow it to be overridden if necessary?

clamoriniere replied:
to my knowledge, this is the standard path and it is used by every manager solution and installer.
I prefer to have someone that opens an issue to add another parameter.

johnkost commented on stable/datadog/templates/cluster-agent-deployment.yaml:
```diff
@@ -78,25 +77,29 @@ spec:
               secretKeyRef:
                 name: {{ template "datadog.appKeySecretName" . }}
                 key: app-key
+                optional: true
```
Should this be optional? Isn't it required for all communication back with DataDog

clamoriniere replied:
it is require if the user want the "external-metrics" feature, and the kubernetes events check.
Else it is not.

johnkost replied:
Ah got it. Thanks for the clarification

johnkost reviewed (commented):
Thanks for this PR! Just some thoughts from a consumer of this chart 😃

clamoriniere committed:
update after review

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

clamoriniere committed:
return error message if apiKey is not provided

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>

L3n41c committed:
Mount the directory containing the CRI socket instead of the socket itself (#5)

This is to handle the cases where the docker daemon is restarted.
In this case, the docker daemon will recreate its docker socket and,
if the container bind-mounted directly the socket, the container would
still have access to the old socket instead of the one of the new docker
daemon.

Signed-off-by: cedric lamoriniere <cedric.lamoriniere@datadoghq.com>
